### PR TITLE
feat: accept a slice of initiators in Publishvolume and ChooseLUN

### PR DESCRIPTION
**Pull request by David-T-White**
Change initiator parameter type to a slice of strings in PublishVolume and ChooseLUN, making it easier to map volumes to multiple initiators

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -432,23 +435,28 @@ func (client *Client) PublishVolume(volumeId string, initiatorName string) (stri
...
 	}
 
 	klog.Infof("using LUN %d", lun)
 
-	if err = client.mapVolumeProcess(volumeId, initiatorName, lun); err != nil {
-		return "", err
+	for _, initiator := range initiators {
+		//TODO optimize this for fewer API calls
+		if err = client.mapVolumeProcess(volumeId, initiator, lun); err != nil {
+			return "", err
```
Comment (seagate-chris) [inline api.go:455]: Shouldn't we just log this error and keep trying to map the rest of the initiators?

File: api.go [hunk 2]
```diff
@@ -432,23 +435,28 @@ func (client *Client) PublishVolume(volumeId string, initiatorName string) (stri
...
+		}
 	}
 
-	klog.Infof("successfully mapped volume (%s) for initiator (%s) using LUN (%d)", volumeId, initiatorName, lun)
+	klog.Infof("successfully mapped volume (%s) for initiator (%s) using LUN (%d)", volumeId, initiators, lun)
```
Comment (seagate-chris) [inline api.go:459]: This would have to be modified to handle mix of successful and failed initiator mappings.

File: api.go (hunk 1 above)
Comment (David-T-White) [reply api.go]: made changes to address this and below comment

**Review by seagate-chris — APPROVED** [review]
Looks good!

File: api.go [hunk 3]
```diff
@@ -314,46 +314,49 @@ func (client *Client) nextLUN(maps InitiatorMapInfo) (int, error) {
...
-		return 1, nil
-	}
-	if err != nil {
-		return -1, err
+
+	var allvolumes []Volume
+	for _, initiatorName := range initiators {
+		volumes, responseStatus, err := client.ShowHostMaps(initiatorName)
+		if err != nil {
+			klog.Infof("error looking for host maps for initiator %s: %s", initiatorName, err)
```
Comment (jskazinski) [inline api.go:324]:
```suggestion
			klog.Errorf("error looking for host maps for initiator %s: %s", initiatorName, err)
```

File: api.go [hunk 4]
```diff
@@ -432,24 +435,35 @@ func (client *Client) PublishVolume(volumeId string, initiatorName string) (stri
...
 	}
 
 	klog.Infof("using LUN %d", lun)
 
-	if err = client.mapVolumeProcess(volumeId, initiatorName, lun); err != nil {
-		return "", err
+	mappingSuccessful := false
+	for _, initiator := range initiators {
+		if err = client.mapVolumeProcess(volumeId, initiator, lun); err != nil {
+			klog.Infof("error mapping volume (%s) for initiator (%s) using LUN (%d): %v", volumeId, initiators, lun, err)
```
Comment (jskazinski) [inline api.go:455]:
```suggestion
			klog.Errorf("error mapping volume (%s) for initiator (%s) using LUN (%d): %v", volumeId, initiators, lun, err)
```

**Review by jskazinski — APPROVED** [review]
Looks good. Two minor suggestions to print an Error message tag versus an Info message tag.

**Pull request merged**